mlflow/projects/databricks.py

```diff
@@ -220,8 +235,12 @@ def _run_shell_command_job(self, project_uri, command, env_vars, cluster_spec):
         # Check syntax of JSON - if it contains libraries and new_cluster, pull those out
         if "new_cluster" in cluster_spec:
             # Libraries are optional, so we don't require that this be specified
-            if "libraries" in cluster_spec:
-                libraries.extend(cluster_spec["libraries"])
+            original_libraries = cluster_spec.get("libraries", [])
+            libraries = (
+                original_libraries
+                if _contains_mlflow_git_uri(original_libraries)
```

Comment:
Can we add a comment explaining what's going on? It seems like, if the cluster spec already includes an MLflow Git URI, then we don't use the PyPI MLflow library definitions above. This ensures that we can install dev versions of MLflow in the Job. Can we also rename `libraries` to `mlflow_libraries`?

Reply:
Yeah, this is only for development purposes:

1. To install a dev version of MLflow in the job cluster.
2. To avoid having two pip requirements for MLflow.

I'll add a comment.